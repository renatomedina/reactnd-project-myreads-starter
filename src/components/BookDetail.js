import React from 'react';
import Header from './Header'
import { Link } from 'react-router-dom'

export default ({ book }) => {
    const { title = "",
        authors = [],
        description = "",
        imageLinks = {},
    } = book

    return (
        <div>
            <Link
                to="/"
                className="close-search"
            >Close</Link>
            <Header
                title="My Reads"
            />
            <div className="book-detail">
                <div className="book-cover" style={{ width: 140, height: 200, backgroundImage: `url(${imageLinks.thumbnail})` }}></div>
            </div>
            <div className="book-title-detail">{title}</div>
            <div className="book-authors.detail">{authors}</div>
            <div className="book-description-detail">{description}</div>
        </div>
    )
}