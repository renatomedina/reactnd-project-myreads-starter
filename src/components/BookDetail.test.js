import React from 'react';
import { shallow, mount } from 'enzyme';
import BookDetail from './BookDetail';
import { MemoryRouter } from 'react-router-dom';
import book from '../__mocks__/BookItem'

describe('<BookDetail />', () => {
    const subjectShallow = () => {
        const b = book
        b.book = book
        return shallow(<BookDetail book={book} />)
    }

    const subjectMount = () => {
        return mount(<MemoryRouter><BookDetail book={book} /></MemoryRouter>)
    }

    it('shallow renders correctly', () => {
        expect(subjectShallow()).toMatchSnapshot()
    })

    it('mount renders correctly', () => {
        expect(subjectMount())
    })

})