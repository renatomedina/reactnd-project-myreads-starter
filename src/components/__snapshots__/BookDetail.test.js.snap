// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookDetail /> shallow renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    book={
        Object {
            "allowAnonLogging": true,
            "authors": Array [
              "William E. Shotts, Jr.",
            ],
            "averageRating": 4,
            "book": [Circular],
            "canonicalVolumeLink": "https://market.android.com/details?id=book-nggnmAEACAAJ",
            "categories": Array [
              "COMPUTERS",
            ],
            "contentVersion": "1.2.2.0.preview.2",
            "description": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \\"shell shock,\\" you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's \\"Evolution of a SysAdmin\\"",
            "id": "nggnmAEACAAJ",
            "imageLinks": Object {
              "smallThumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
              "thumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
            },
            "industryIdentifiers": Array [
              Object {
                "identifier": "9781593273897",
                "type": "ISBN_13",
              },
              Object {
                "identifier": "1593273894",
                "type": "ISBN_10",
              },
            ],
            "infoLink": "https://play.google.com/store/books/details?id=nggnmAEACAAJ&source=gbs_api",
            "language": "en",
            "maturityRating": "NOT_MATURE",
            "pageCount": 480,
            "panelizationSummary": Object {
              "containsEpubBubbles": false,
              "containsImageBubbles": false,
            },
            "previewLink": "http://books.google.com/books?id=nggnmAEACAAJ&dq=linux&hl=&cd=3&source=gbs_api",
            "printType": "BOOK",
            "publishedDate": "2012",
            "publisher": "No Starch Press",
            "ratingsCount": 2,
            "readingModes": Object {
              "image": false,
              "text": true,
            },
            "shelf": "currentlyReading",
            "subtitle": "A Complete Introduction",
            "title": "The Linux Command Line",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="close-search"
          replace={false}
          to="/"
>
          Close
</Link>,
        <Unknown
          title="My Reads"
/>,
        <div
          className="book-detail"
>
          <div
                    className="book-cover"
                    style={
                              Object {
                                        "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
                                        "height": 200,
                                        "width": 140,
                                      }
                    }
          />
</div>,
        <div
          className="book-title-detail"
>
          The Linux Command Line
</div>,
        <div
          className="book-authors.detail"
>
          William E. Shotts, Jr.
</div>,
        <div
          className="book-description-detail"
>
          You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial "shell shock," you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's "Evolution of a SysAdmin"
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Close",
          "className": "close-search",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Close",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "My Reads",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="book-cover"
            style={
                        Object {
                                    "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
                                    "height": 200,
                                    "width": 140,
                                  }
            }
/>,
          "className": "book-detail",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "book-cover",
            "style": Object {
              "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
              "height": 200,
              "width": 140,
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Linux Command Line",
          "className": "book-title-detail",
        },
        "ref": null,
        "rendered": "The Linux Command Line",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "William E. Shotts, Jr.",
          ],
          "className": "book-authors.detail",
        },
        "ref": null,
        "rendered": Array [
          "William E. Shotts, Jr.",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \\"shell shock,\\" you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's \\"Evolution of a SysAdmin\\"",
          "className": "book-description-detail",
        },
        "ref": null,
        "rendered": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \\"shell shock,\\" you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's \\"Evolution of a SysAdmin\\"",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="close-search"
            replace={false}
            to="/"
>
            Close
</Link>,
          <Unknown
            title="My Reads"
/>,
          <div
            className="book-detail"
>
            <div
                        className="book-cover"
                        style={
                                    Object {
                                                "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
                                                "height": 200,
                                                "width": 140,
                                              }
                        }
            />
</div>,
          <div
            className="book-title-detail"
>
            The Linux Command Line
</div>,
          <div
            className="book-authors.detail"
>
            William E. Shotts, Jr.
</div>,
          <div
            className="book-description-detail"
>
            You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial "shell shock," you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's "Evolution of a SysAdmin"
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Close",
            "className": "close-search",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Close",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "My Reads",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="book-cover"
              style={
                            Object {
                                          "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
                                          "height": 200,
                                          "width": 140,
                                        }
              }
/>,
            "className": "book-detail",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "book-cover",
              "style": Object {
                "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
                "height": 200,
                "width": 140,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Linux Command Line",
            "className": "book-title-detail",
          },
          "ref": null,
          "rendered": "The Linux Command Line",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "William E. Shotts, Jr.",
            ],
            "className": "book-authors.detail",
          },
          "ref": null,
          "rendered": Array [
            "William E. Shotts, Jr.",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \\"shell shock,\\" you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's \\"Evolution of a SysAdmin\\"",
            "className": "book-description-detail",
          },
          "ref": null,
          "rendered": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \\"shell shock,\\" you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's \\"Evolution of a SysAdmin\\"",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
